projects/fw_103_hw/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>
+#include <stdlib.h>
+
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress pot_address = {
+  .port = GPIO_PORT_A,
+  .pin = 6,
+};
+
+static GpioAddress button_address = {
+  .port = GPIO_PORT_B,
+  .pin = 2,
+};
+
+static GpioSettings pot_settings = {
```

Comment:
Generally we define these settings only in the function which we use them (main) to avoid cluttering the program. If you use them in more than one place in the program, like the GPIO addresses, then this is ok.